currency/code.go

```diff
@@ -336,3 +336,8 @@ func (c Code) IsCryptocurrency() bool {
 func (c Code) IsStableCurrency() bool {
 	return c.Item != nil && c.Item.Role == Stable
 }
+
+// Currency allows an item to revert to a code
+func (i *Item) Currency() Code {
+	return NewCode(i.Symbol)
```

Comment:
Nice! Can we add a test. Just for added protection:
```go
if i == nil {
		return EMPTYCODE
}
```

Suggestion Add specific method for Upper and lower casing. Using field i.Lower.